StereoKit/Systems/Renderer.cs

```diff
@@ -284,6 +291,73 @@ public static void Screenshot(Vec3 from, Vec3 at, int width, int height, string
 		public static void Screenshot(string filename, Vec3 from, Vec3 at, int width, int height, float fieldOfViewDegrees = 90)
 			=> NativeAPI.render_screenshot(filename, from, at, width, height, fieldOfViewDegrees);
 
+		/// <summary>Schedules a screenshot for the end of the frame! The view
+		/// will be rendered from the given position at the given point, with a
+		/// resolution the same size as the screen's surface. This overload
+		/// allows for retrieval of the color data directly from the render
+		/// thread! You can use the color data directly by saving/processing it
+		/// inside your callback, or you can keep the data alive for as long as
+		/// it is referenced.</summary>
+		/// <param name="onScreenshot">Outputs a reference to the color data
+		/// and its length which represent the current scene from a requested
+		/// viewpoint.</param>
+		/// <param name="from">Viewpoint location.</param>
+		/// <param name="at">Direction the viewpoint is looking at.</param>
+		/// <param name="width">Size of the screenshot horizontally, in pixels.
+		/// </param>
+		/// <param name="height">Size of the screenshot vertically, in pixels.
+		/// </param>
+		/// <param name="fieldOfViewDegrees">The angle of the viewport, in 
+		/// degrees.</param>
+		public static void Screenshot(ScreenshotCallback onScreenshot, Vec3 from, Vec3 at, int width, int height, float fieldOfViewDegrees = 90)
+		{
+			if (_renderCaptureCallbacks is null) _renderCaptureCallbacks = new ConcurrentQueue<RenderOnScreenshotCallback>();
```

Comment:
I hadn't actually seen the matching form of the `is` keyword before, no action needed here, but I'm glad you've brought it to my attention because I rather like how it works!

Reply:
I really like them too - they're readable and make the ReferenceEquals intention clear. Also, when you update to a newer C# language version, you can later null-coalescing assignment ??= (which also uses the "is" check) for this line too :)